### DISCUSSION TITLE
Version 2.1.2

### DISCUSSION THREAD

**Discussion by agoose77** [scikit-hep/awkward#2381] [Deprecations]
## New features

* feat: use cppyy for JIT by @ianna in https://github.com/scikit-hep/awkward/pull/2306
* feat!: rename cpptype by @agoose77 in https://github.com/scikit-hep/awkward/pull/2331
* feat: implemented GrowableBuffer in Numba as a start toward LayoutBuilder by @jpivarski in https://github.com/scikit-hep/awkward/pull/2349
* feat: add `to_backend` to `ak.record.Record` by @agoose77 in https://github.com/scikit-hep/awkward/pull/2355
* feat: add type equality `is_equal_to` member by @agoose77 in https://github.com/scikit-hep/awkward/pull/2368

## Bug-fixes and performance

* fix: expose array interface for CUDA by @agoose77 in https://github.com/scikit-hep/awkward/pull/2327
* fix: test untested path in `ak.cartesian` & broadcasting by @agoose77 in https://github.com/scikit-hep/awkward/pull/2329
* fix: `ak.cartesian` for typetracer by @agoose77 in https://github.com/scikit-hep/awkward/pull/2295
* fix: Numba string reference count memory leak. by @jpivarski in https://github.com/scikit-hep/awkward/pull/2332
* fix: generate an array view when an Array C++ type is requested by @ianna in https://github.com/scikit-hep/awkward/pull/2335
* fix: do not expose an `rdfentry_` column by @ianna in https://github.com/scikit-hep/awkward/pull/2343
* fix: generate RDataSource API based on ROOT attribute by @ianna in https://github.com/scikit-hep/awkward/pull/2345
* fix: simplify depth limit test to support bigger-than-depth values by @agoose77 in https://github.com/scikit-hep/awkward/pull/2347
* fix: use cppyy include by @ianna in https://github.com/scikit-hep/awkward/pull/2348
* fix: convert ufunc arguments to same backend by @agoose77 in https://github.com/scikit-hep/awkward/pull/2354
* fix: support N-d sequences in `TypeTracer.asarray` by @agoose77 in https://github.com/scikit-hep/awkward/pull/2361
* fix: make `from_iter` require iterables! by @agoose77 in https://github.com/scikit-hep/awkward/pull/2353
* fix: avoid stack overflow with std::unique_ptr linked list by @ianna in https://github.com/scikit-hep/awkward/pull/2366
* fix: `ak.unzip` visits all contents by @agoose77 in https://github.com/scikit-hep/awkward/pull/2373
* fix: remove spurious typetracer conversion by @agoose77 in https://github.com/scikit-hep/awkward/pull/2375

## Other

* refactor: hide `awkward.typing` by @agoose77 in https://github.com/scikit-hep/awkward/pull/2324
* refactor: move `_parameters_XXX` functions into `_parameters` submodule by @agoose77 in https://github.com/scikit-hep/awkward/pull/2325
* refactor: move `merge_as_union` to `ak_concatenate` by @agoose77 in https://github.com/scikit-hep/awkward/pull/2351
* refactor: improve broadcasting logic readability by @agoose77 in https://github.com/scikit-hep/awkward/pull/2359
* refactor: wrap exceptions at catch time by @agoose77 in https://github.com/scikit-hep/awkward/pull/2370
* docs: fix pyodide (again!) by @agoose77 in https://github.com/scikit-hep/awkward/pull/2326
* docs: add import statement by @raybellwaves in https://github.com/scikit-hep/awkward/pull/2358
* docs: update mybinder link to example notebook by @raybellwaves in https://github.com/scikit-hep/awkward/pull/2357
* docs: fix extra for pyodide by @agoose77 in https://github.com/scikit-hep/awkward/pull/2360
* docs: add raybellwaves as a contributor for doc by @allcontributors in https://github.com/scikit-hep/awkward/pull/2379
* chore: update pre-commit hooks by @pre-commit-ci in https://github.com/scikit-hep/awkward/pull/2313
* chore: update pre-commit hooks by @pre-commit-ci in https://github.com/scikit-hep/awkward/pull/2328

## New Contributors
* @raybellwaves made their first contribution in https://github.com/scikit-hep/awkward/pull/2358

**Full Changelog**: https://github.com/scikit-hep/awkward/compare/v2.1.1...v2.1.2

<hr /><em>This discussion was created from the release <a href='https://github.com/scikit-hep/awkward/releases/tag/v2.1.2'>v2.1.2</a>.</em>

**Comment by jpivarski** [comment]
Hi @agoose77,

Two small corrections:

  1. This release needed to be named "Version 2.1.2", not "v2.1.2" (that's the git tag name). [It did deploy successfully](https://pypi.org/project/awkward/), which I didn't think would happen, but maybe that only depends on the git tag name, not the release name. I fixed the title, both in the release and the discussion, so there's no action item.
  2. It's missing `header-only.zip`. I don't know how it lost that—maybe it depends on release name? I didn't fix this (I don't know how), so it should probably be done manually. Is it usually done manually? If so, it has to become automatic so that it's not forgotten...

**Reply by agoose77** [reply]
Thanks! I did rename the release, I wonder what happened. I did, however, forget the headers. We need to extend our CI for this.
